flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/enumerator/PulsarSourceEnumerator.java

```diff
@@ -220,11 +223,14 @@ private void createSubscription(List<TopicPartition> newPartitions) {
             CursorPosition position =
                     startCursor.position(partition.getTopic(), partition.getPartitionId());
 
-            if (sourceConfiguration.isResetSubscriptionCursor()) {
-                sneakyAdmin(() -> position.seekPosition(pulsarAdmin, topic, subscriptionName));
-            } else {
-                sneakyAdmin(
-                        () -> position.createInitialPosition(pulsarAdmin, topic, subscriptionName));
+            try {
+                if (sourceConfiguration.isResetSubscriptionCursor()) {
+                    position.seekPosition(pulsarAdmin, topic, subscriptionName);
+                } else {
+                    position.createInitialPosition(pulsarAdmin, topic, subscriptionName);
+                }
+            } catch (PulsarAdminException e) {
+                throw new FlinkRuntimeException(e);
```

Comment:
Going up the call hierarchy, this method exception would be exposed (and not caught) in a method that's uses for error handling which sounds strange? Or is it reasonable? :thinking:

Reply:
This method is call in `context.callAsync(this::getSubscribedTopicPartitions, this::checkPartitionChanges);`. It's hard to throw the exception here.

Reply:
My bad, we can still throw this exception will a little more works on code refactoring.